src/main/asciidoc/transactions.adoc

```diff
@@ -33,82 +41,257 @@ static class Config extends AbstractCouchbaseConfiguration {
+be a fully qualified class name or a substring of a fully qualified class name
+for an exception type (which must be a subclass of Throwable), with no
+wildcard support at present. For example, a value of
+"jakarta.servlet.ServletException" or "ServletException" will
+match jakarta.servlet.ServletException and its ubclasses.
+ 
+WARNING: You must carefully consider how specific a pattern
+is and whether to include package information (which isn't mandatory). For example,
+"Exception" will match nearly anything and will probably hide other
+rules. "java.lang.Exception" would be correct if "Exception"
+were meant to define a rule for all checked exceptions. With more unique
+exception names such as "BaseBusinessException" there is likely no
+need to use the fully qualified class name for the exception pattern. Furthermore,
+rollback rules defined via patterns may result in unintentional matches for
+similarly named exceptions and nested classes. This is due to the fact that a
+thrown exception is considered to be a match for a given pattern-based rollback
+rule if the name of thrown exception contains the exception pattern configured
+for the rollback rule. For example, given a rule configured to match against
+"com.example.CustomException", that rule will match against an exception
+named com.example.CustomExceptionV2 (an exception in the same package as
+CustomException but with an additional suffix) or an exception named
+com.example.CustomException$AnotherException (an exception declared as
+a nested class in CustomException).
+ 
+For specific information about the semantics of other attributes in this
+annotation, consult the org.springframework.transaction.TransactionDefinition
+and org.springframework.transaction.interceptor.TransactionAttribute javadocs.
+ 
+=== Transaction Management
+ 
+This annotation commonly works with thread-bound transactions managed by a
+org.springframework.transaction.PlatformTransactionManager, exposing a
+transaction to all data access operations within the current execution thread.
+Note: This does NOT propagate to newly started threads within the method.
+
+Couchbase does not supply a reactive transaction manager, therefore it cannot support
+@Transactional on reactive methods. Couchbase provides reactive transaction support
+via the CouchbaseTransactionalOperator. 
 
 .Transaction Access
 ====
 [source,java]
 ----
-@Autowired
-Transactions transactions;
+// this object must be provided by the Spring framework. By @Autowire or from ac.getBean().
+// If created directly with it's constructor, the method interceptor will not be called.
+@Autowired PersonService personService;
+
+    Person p = personService.savePerson(WalterWhite);
+
 
-@Autowired
-CouchbaseClientFactory couchbaseClientFactory;
+@Service
+@Component
+public class PersonService {
+
+    final CouchbaseOperations personOperations;
+
+    public PersonService(CouchbaseOperations ops) {
```

Comment:
No @Autowired?

Reply:
I'll just assume this does work then and the dependency injection happens even without Autowired.

Reply:
The magic happens with

@Autowired PersonService personService;

spring looks for candidates for the Autowiring.  It finds that there is an @Service PersonService.  It examines the constructor for PersonService and sees that it takes a CouchbaseOperation arg.  It looks for candidates for that, and finds that there is an @Bean CouchbaseTemplate and that CouchbaseTemplate implements CouchbaseOperations - so it uses that.

Reply:
Ah interesting, I didn't know the DI could propagate through like that, e.g. is only required on the initial field/ctor - I thought each constructor (or field) invoked also needed the @Autowired to work.